module/mempool/herocache/backdata/cache.go

```diff
@@ -152,105 +152,105 @@ func NewCache(
 	return bd
 }
 
-// Has checks if backdata already contains the entity with the given identifier.
-func (c *Cache) Has(entityID flow.Identifier) bool {
+// Has checks if backdata already contains the value with the given identifier.
+func (c *Cache[V]) Has(key flow.Identifier) bool {
 	defer c.logTelemetry()
 
-	_, _, _, ok := c.get(entityID)
+	_, _, _, ok := c.get(key)
 	return ok
 }
 
-// Add adds the given entity to the backdata and returns true if the entity was added or false if
-// a valid entity already exists for the provided ID.
-func (c *Cache) Add(entityID flow.Identifier, entity flow.Entity) bool {
+// Add adds the given value to the backdata and returns true if the value was added or false if
+// a valid value already exists for the provided ID.
+func (c *Cache[V]) Add(key flow.Identifier, value V) bool {
 	defer c.logTelemetry()
-	return c.put(entityID, entity)
+	return c.put(key, value)
 }
 
-// Remove removes the entity with the given identifier and returns the removed entity and true if
-// the entity was removed or false if the entity was not found.
-func (c *Cache) Remove(entityID flow.Identifier) (flow.Entity, bool) {
+// Remove removes the value with the given identifier and returns the removed value and true if
+// the value was removed or false if the value was not found.
+func (c *Cache[V]) Remove(key flow.Identifier) (value V, ok bool) {
 	defer c.logTelemetry()
 
-	entity, bucketIndex, sliceIndex, exists := c.get(entityID)
+	value, bucketIndex, sliceIndex, exists := c.get(key)
 	if !exists {
-		return nil, false
+		return value, false
 	}
 	// removes value from underlying entities list.
-	c.invalidateEntity(bucketIndex, sliceIndex)
+	c.invalidateValue(bucketIndex, sliceIndex)
 
 	// frees up slot
 	c.unuseSlot(bucketIndex, sliceIndex)
 
 	c.collector.OnKeyRemoved(c.entities.Size())
-	return entity, true
+	return value, true
 }
 
-// Adjust adjusts the entity using the given function if the given identifier can be found.
-// Returns a bool which indicates whether the entity was updated as well as the updated entity.
-func (c *Cache) Adjust(entityID flow.Identifier, f func(flow.Entity) flow.Entity) (flow.Entity, bool) {
+// Adjust adjusts the value using the given function if the given identifier can be found.
+// Returns a bool which indicates whether the value was updated as well as the updated value.
+func (c *Cache[V]) Adjust(key flow.Identifier, f func(V) V) (value V, ok bool) {
 	defer c.logTelemetry()
 
-	entity, removed := c.Remove(entityID)
+	value, removed := c.Remove(key)
 	if !removed {
-		return nil, false
+		return value, false
 	}
 
-	newEntity := f(entity)
-	newEntityID := newEntity.ID()
+	newValue := f(value)
 
-	c.put(newEntityID, newEntity)
+	// TODO(malleability, #7074): Think of a better solution cause of removing and inserting value with same ID https://github.com/onflow/flow-go/issues/7074
```

Comment:
Thanks for pointing this out. From my perspective the `Adjust` and `AdjustWithInit` methods will be very easy to misuse under the new semantics, where we don't re-compute the ID of the stored value. 

In addition to the confusing semantics, they also have very limited value. Anything you can do with `Adjust` you can also do with multiple calls to `Add` and `Remove` while holding the lock protecting the Cache.

It also seems like these methods are only used in tests. So I suggest we remove both methods. (`Cache` will no longer implement `mempool.Backdata`... But I think that is OK.)

Reply:
I think there might be an issue with this approach, I want to discuss this so we can reach a common denominator.